lib/model.js

```diff
@@ -231,20 +231,36 @@ Model.prototype.$__handleSave = function(options, callback) {
   if ('safe' in options) {
     _handleSafe(options);
   }
+
   applyWriteConcern(this.$__schema, options);
-  if ('w' in options) {
-    saveOptions.w = options.w;
-  }
-  if ('j' in options) {
-    saveOptions.j = options.j;
-  }
-  if ('wtimeout' in options) {
-    saveOptions.wtimeout = options.wtimeout;
-  }
-  if ('checkKeys' in options) {
-    saveOptions.checkKeys = options.checkKeys;
+  if (typeof options.writeConcern != 'undefined') {
+    if ('w' in options.writeConcern) {
+      saveOptions.w = options.writeConcern.w;
+    }
+    if ('j' in options.writeConcern) {
+      saveOptions.j = options.writeConcern.j;
+    }
+    if ('wtimeout' in options.writeConcern) {
+      saveOptions.wtimeout = options.writeConcern.wtimeout;
+    }
+    if ('checkKeys' in options.writeConcern) {
+      saveOptions.checkKeys = options.writeConcern.checkKeys;
+    }
+    saveOptions = { writeConcern: saveOptions };
```

Comment:
This really should be `saveOptions.writeConcern = options.writeConcern` or something like that. Why overwrite all `saveOptions`?

Reply:
You really shouldn't be overwriting `saveOptions` here. You'd be removing all other top-level `saveOptions`, like `session`. Instead, do something like:

```javascript
saveOptions.writeConcern = Object.assign({}, options.writeConcern);
```